Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -4581,17 +4582,26 @@ void MediaPlayerPrivateGStreamer::checkPlayingConsistency()
     }
 }
 
-static void applyAudioSinkDevice(GstElement* audioSinkBin, const String& deviceId)
+bool MediaPlayerPrivateGStreamer::applyAudioSinkDevice(GstElement* audioSink, GstDevice* device)
 {
-    for (auto* element : GstIteratorAdaptor<GstElement>(gst_bin_iterate_sinks(GST_BIN_CAST(audioSinkBin)))) {
-        // pulsesink and alsasink have a "device" property, whilst pipewiresink has "target-object"
-        if (gstElementMatchesFactoryAndHasProperty(element, "pulsesink"_s, "device"_s) || gstElementMatchesFactoryAndHasProperty(element, "alsasink"_s, "device"_s))
-            g_object_set(element, "device", deviceId.utf8().data(), nullptr);
-        else if (gstElementMatchesFactoryAndHasProperty(element, "pipewiresink"_s, "target-object"_s))
-            g_object_set(element, "target-object", deviceId.utf8().data(), nullptr);
-        else if (GST_IS_BIN(element))
-            applyAudioSinkDevice(element, deviceId);
+    bool changed = false;
+
+    if (GST_IS_BIN(audioSink)) {
+        for (auto* element : GstIteratorAdaptor<GstElement>(gst_bin_iterate_sinks(GST_BIN_CAST(audioSink)))) {
+            if (applyAudioSinkDevice(element, device))
+                changed = true;
+        }
+        return changed;
     }
+
+    if (gstElementFactoryEquals(audioSink, "fakesink"_s)) {
+        GST_WARNING_OBJECT(pipeline(), "Skipped unexpected fakesink, your audio configuration may have issues.");
+        return changed;
+    }
+
```

Comment:
`fakeaudiosink` might need to be handled here too, it's upranked on post-commit bots running layout tests.

Reply:
Do you have any suggestion about how to handle it? Maybe we can keep this code for non-developer builds, and for developer builds consider that the audio output was done if we reach a fake sink?

Reply:
Also I think the condition here needs to be changed check for `fakeaudiosink` (instead of `fakesink` which AFAICS is for video).

Reply:
> Do you have any suggestion about how to handle it? 

Maybe just a GST_DEBUG and return early is sufficient.

> Also I think the condition here needs to be changed check for fakeaudiosink (instead of fakesink which AFAICS is for video).

I don't think so, `autoaudiosink` and `autovideosink` create a `fakesink` as fallback (see `gst_auto_detect_create_fake_element_default()`), so the existing check on `fakesink` looks appropriate.

Reply:
Yes, because as mentioned above, `autoaudiosink` falls back to `fakesink` if no other sink was found.

Reply:
Done, thanks again!